kiota_abstractions/request_information.py

```diff
@@ -88,14 +88,29 @@ def url(self, url: Url) -> None:
 
     @property
     def request_headers(self) -> Optional[Dict]:
-        return self.headers
-
-    def add_request_headers(self, headers_to_add: Optional[Dict[str, str]]) -> None:
+        final = {}
+        for key, value in self.headers.items():
+            # Use set to remove duplicates
+            final[key] = ', '.join(list(set(value)))
+        return final
+
+    def add_request_headers(
+        self, headers_to_add: Optional[Dict[str, Union[str, List[str]]]]
+    ) -> None:
         """Adds headers to the request
         """
         if headers_to_add:
-            for key in headers_to_add:
-                self.headers[key.lower()] = headers_to_add[key]
+            for key, value in headers_to_add.items():
+                if key.lower() in self.headers:
```

Comment:
Could move 
```python
key.lower()
```
to a variable instead of lowering every time.

Reply:
https://github.com/microsoft/kiota-abstractions-python/pull/39/commits/4d22efd4a9bdeae2be2067e9deb20e873d4631bb